packages/block-library/src/buttons/edit.native.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+/**
+ * WordPress dependencies
+ */
+import {
+	InnerBlocks,
+	__experimentalAlignmentHookSettingsProvider as AlignmentHookSettingsProvider,
+} from '@wordpress/block-editor';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose, useResizeObserver } from '@wordpress/compose';
+import { createBlock } from '@wordpress/blocks';
+import { useState, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { name as buttonBlockName } from '../button/';
+import styles from './editor.scss';
+
+const ALLOWED_BLOCKS = [ buttonBlockName ];
+const BUTTONS_TEMPLATE = [ [ 'core/button' ] ];
+
+function ButtonsEdit( {
+	isSelected,
+	attributes,
+	onDelete,
+	onAddNextButton,
+	shouldDelete,
+	isSelectedButton,
+} ) {
+	const { align } = attributes;
+	const [ resizeObserver, sizes ] = useResizeObserver();
+	const [ maxWidth, setMaxWidth ] = useState( 0 );
+	const shouldRenderFooterAppender = isSelected || isSelectedButton;
+	const { marginLeft: spacing } = styles.spacing;
+
+	useEffect( () => {
+		const { width } = sizes || {};
+		const margins = 2 * styles.parent.marginRight;
+
+		setMaxWidth( width - margins );
```

Comment:
At line 41 you added a `|| {}` to be sure that the app doesn't crash when the `sizes` are `undefined`. In that case, the `width` will be`undefined` and the maxWidth will be set to `NaN` because i.e `undefined - 10 = NaN`